### DISCUSSION TITLE
Question regarding media versions

### DISCUSSION THREAD

**Discussion by fnagel** [sulu/sulu#7622] [Q&A] [answered]
Hi there, Sulu community!

I have a question regarding media versions. When someone upload the same file multiple times, multiple versions are created. I can watch and manage those in the "history" tab of the media item.

So, when there are two version (both created from the same file) and now someone deletes "version 1", the original file of "version 2" is no longer available too (500 error, in logs its a "FileNotFoundException" error). 

Why is that? That does not make sense to me. I assumed version of a media item are independent and old version could be deleted without removing the current version in use. I missing something here?

**Comment by alexander-schranz** [comment, marked as answer]
@fnagel Really sorry I missed this question.

Two Versions can not reference the same "Physical" File. If you upload a file `photo.jpg` and upload the same file it the second file will be postfixed with `photo-1.jpg`. This is handled in the `getUniqueFilename` method in the Sulu storage: https://github.com/sulu/sulu/blob/15a676466de29d815d016b37dd7cb97fef1cc76d/src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php#L122

So if version 1 is removed only `photo.jpg` is removed if version 2 is removed only `photo-1.jpg` is removed.

The only thing which could your error make happen if somebody overwrite the storage or did write set specific storageoptions themselve by creating the media files version entity by there own. But in normal Sulu way this should never happening.
